shared/src/vyxal/Elements.scala

```diff
@@ -2386,10 +2589,130 @@ object Elements:
         VList.from(ListHelpers.map(a, iter).vzip(iter))
       case (a, b) =>
         ListHelpers.makeIterable(a).vzip(ListHelpers.makeIterable(b))
+    },
+    addDirect(
+      "£",
+      "Set Register",
+      List("set-register", "->register", "set-reg", "->reg"),
+      Some(1),
+      "a: any -> register = a",
+    ) { ctx ?=>
+      ctx.globals.register = ctx.pop()
+    },
+    addDirect(
+      "¥",
+      "Get Register",
+      List("get-register", "get-reg", "register", "<-register", "<-reg"),
+      None,
+      " -> push the value of the register",
+    ) { ctx ?=>
+      ctx.push(ctx.globals.register)
+    },
+    addDirect(
+      "←",
+      "Rotate Stack Left",
+      List("rotate-stack-left"),
+      None,
+      " -> rotate the entire stack left once",
+    ) { ctx ?=>
+      ctx.rotateLeft
+    },
+    addDirect(
+      "→",
+      "Rotate Stack Right",
+      List("rotate-stack-right"),
+      None,
+      " -> rotate the entire stack right once",
+    ) { ctx ?=>
+      ctx.rotateRight
+    },
+    addDirect(
+      "\\",
+      "Dump",
+      List("dump"),
+      Some(1),
+      "a: any -> dump all values on the stack",
+    ) { ctx ?=>
+      val a = ListHelpers.makeIterable(ctx.pop())
+      for v <- a do ctx.push(v)
+    },
+    addPart(
+      Monad,
+      "†",
+      "Length of Consecutive Groups",
+      List("len-consecutive", "gvl", "gavel"),
+      false,
+      "a: any -> lengths of consecutive groups of a",
+    ) {
+      case a =>
+        val iterable = ListHelpers.makeIterable(a)
+        val groups = ListHelpers.groupConsecutive(iterable)
+        VList.from(groups.map(ListHelpers.makeIterable(_).length))
+    },
+    addPart(
+      Monad,
+      "Π",
+      "Product",
+      List("product", "prod"),
+      false,
+      "a: lst -> product of a",
+    ) {
+      case a: VList => ListHelpers.product(a)
+      case a: String => ListHelpers.product(ListHelpers.makeIterable(a))
+      case a: VNum => ListHelpers.product(ListHelpers.makeIterable(a))
```

Comment:
Is this just factorial?

Reply:
It's neither.

`ListHelpers.makeIterable(a)` returns a list of digits by default, not a range.